gym_xiangqi/envs/xiangqi_env.py

```diff
@@ -21,5 +178,278 @@ def render(self, mode='human'):
     def close(self):
         pass
 
+    def init_pieces(self):
+        """
+        Method initializes and stores all ally and enemy pieces
+        """
+        # initialize agent and enemy pieces
+        for r in range(self.shape[0]):
+            for c in range(self.shape[1]):
+                piece_id = self.initial_board[r][c]
+                init = self.id_to_class[abs(piece_id)]
+                if piece_id < 0:
+                    self.enemy_piece[-piece_id] = init(self.enemy_color, r, c)
+                elif piece_id > 0:
+                    self.agent_piece[piece_id] = init(self.agent_color, r, c)
+
+    def move_to_action_space(self, piece_id, start, end):
+        """
+        The action space is a 1D flat array. We can convert piece id,
+        start position and end position to a corresponding index value
+        in the action space.
+
+        Parameters:
+            piece_id (int): a piece ID integer
+            start (tuple(int)): (row, col) start coordinate
+            end (tuple(int)): (row, col) end coordinate
+        return:
+            Index within the self.possible_actions
+        """
+        piece_id_val = (piece_id-1) * pow(self.total_pos, 2)
```

Comment:
Hmm is there a specific reason to have the 1D flat array to represent action? Feels like the array is really big. Is it possible to just use "move" instead, which is `(id, start, end)`.

Reply:
The size of the array is (9*10)^2 * 16 is it?

Reply:
We decided to keep this for now